pkg/pipeline/conntrack/conntrack.go

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (C) 2022 IBM, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package conntrack
+
+import (
+	"encoding/hex"
+	"fmt"
+	"hash"
+	"hash/fnv"
+
+	"github.com/netobserv/flowlogs-pipeline/pkg/api"
+	"github.com/netobserv/flowlogs-pipeline/pkg/config"
+	log "github.com/sirupsen/logrus"
+)
+
+// direction indicates the direction of a flow log in a connection. It's used by aggregators to determine which split
+// of the aggregator should be updated, xxx_AB or xxx_BA.
+type direction uint8
+
+const (
+	dirNA direction = iota
+	dirAB
+	dirBA
+)
+
+type ConnectionTracker interface {
+	Track(flowLogs []config.GenericMap) []config.GenericMap
+}
+
+type conntrackImpl struct {
+	config api.ConnTrack
+	hasher hash.Hash
+	// TODO: should the key of the map be a custom hashStrType instead of string?
+	hash2conn                 map[string]connection
+	aggregators               []aggregator
+	shouldOutputFlowLogs      bool
+	shouldOutputNewConnection bool
+}
+
+func (ct *conntrackImpl) Track(flowLogs []config.GenericMap) []config.GenericMap {
+	log.Debugf("Entering Track")
+	log.Debugf("Track none, in = %v", flowLogs)
+
+	var outputRecords []config.GenericMap
+	for _, fl := range flowLogs {
+		// TODO: think of returning a string rather than []byte
+		computedHash, err := ComputeHash(fl, ct.config.KeyDefinition, ct.hasher)
+		if err != nil {
+			log.Warningf("skipping flow log: %v", err)
```

Comment:
Can you add also the flowlog to the warning --- so we will be able to debug and understand what is broken